### DISCUSSION TITLE
Can docking use ArUco marker, or must be April Tag?

### DISCUSSION THREAD

**Discussion by slowrunner** [ros-navigation/navigation2#4799] [Q&A] [answered]
My dock has an ArUco marker, and I just tested aruco_OpenCV is “seeing” the tag.  I haven’t calibrated yet though so the x,y is off.

can the nav2 docking feature use ArUco marker or do I need to put an April tag on my dock?

Does the back onto dock parameter work for camera on front, contacts on rear and AprilTag on the dock not visible during the backing?  If I get the bot to a 1cm 2deg heading tolerance, I cad add the 180 spin to my /dock action.

**Comment by SteveMacenski** [comment, marked as answer]
There is no requirement to use any kind of detection solution, you can use whatever makes sense for your applications, including custom camera detectors, AI, QR codes, 3D lidar detection, etc. We just show demos using Apriltags since they're commonly used, supported well in ROS 2, and has Nvidia Isaac GPU acceleration for Jetson users. 

> Does the back onto dock parameter work for camera on front, contacts on rear and AprilTag on the dock not visible during the backing? If I get the bot to a 1cm 2deg heading tolerance, I cad add the 180 spin to my /dock action.

Not currently, the sensor has to be in the direction of docking for continuous updates during the docking procedure. There is a PR open trying to implement https://github.com/ros-navigation/navigation2/pull/4749 but it has been stalled by its author recently and I'm sure would be open to someone else helping get it across the line! See the ticket https://github.com/ros-navigation/navigation2/issues/4403
